package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py

```diff
@@ -407,24 +429,35 @@ def guess_hydrogens(self,
             ))
         ]
 
-        hydrogens_list = np.unique(
-            [
-                '(resname {} and name {})'.format(r, p) for r, p in zip(hydrogens_ag.resnames, hydrogens_ag.names)
-            ]
-        )
+        if hasattr(hydrogens_ag,"resnames") and hasattr(hydrogens_ag,"names"):
+            hydrogens_list = np.unique(
+                [
+                    '(resname {} and name {})'.format(r, p) for r, p in zip(hydrogens_ag.resnames, hydrogens_ag.names)
+                ]
+            )
+        else:
+            hydrogens_list = np.unique(
+                [
+                    'type {}'.format(tp) for tp in hydrogens_ag.types
```

Comment:
In most cases an atom type in LAMMPS is all identical with the exception of forcefields such as ReaxFF and COMB. Although you are right that LAMMPS allows for atoms to have individual charges so this *could* be an issue as forcefields develop...

Here's the issue, these structures look for a list of atom selection strings that represent **categories**, so if this was on an index basis, the list would be very long, and the resulting output would be useless (representing a single interaction). One could have this structure instead search the atoms of each type and make a long string that specifies the specific indices that meet the charge criteria. However, this string will be unnecessary computational time in 99% of cases, and in the 1% I would argue it would skew the results as these many-body forcefields will continue to fluctuate in charge throughout the simulation, as they are supposed to. I would argue counting "low" or "high" charges in that case wouldn't disrupt the intention of the calculation, it would result in a broken hydrogen bond with a more "realistic" decay time due to local atom environments.